peercrawler.py

```diff
@@ -82,6 +82,34 @@ def count_good_peers(self):
     def count_peers(self):
         return len(self.get_peers_copy())
 
+    def listen_incoming(self):
+        with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as s:
+            s.bind(("::", 7075))
+            s.listen()
+
+            while True:
+                connection, address = s.accept()
+                threading.Thread(target=self.handle_incoming, args=(connection, address), daemon=True).start()
+
+    def handle_incoming(self, connection: socket.socket, address):
+        while True:
```

Comment:
I would also add the restriction that a keepalive must arrive within one mminute.
Otherwise, give up the socket and close the socket, to stop any cases of lingering sockets.